src/seedu/addressbook/data/employee/UniqueEmployeeList.java

```diff
@@ -75,10 +75,12 @@ public UniqueEmployeeList(UniqueEmployeeList source) {
     }
 
     /**
-     * Checks if the list contains an equivalent employee as the given argument.
+     * Checks if the list contains an employee that already exists in Rms.
      */
-    public boolean contains(ReadOnlyEmployee toCheck) {
-        return employeeInternalList.contains(toCheck);
+    public boolean containsDuplicate(ReadOnlyEmployee toCheck) {
+        return employeeInternalList.stream()
+        .filter(o -> o.getName().toString().toLowerCase()
+                .equals(toCheck.getName().toString().toLowerCase())).findFirst().isPresent();
```

Comment:
Need to improve readability.
Try splitting it into multiple lines/methods.

Reply:
Not possible to condense into lesser methods as it would be too complicated. Instead split it into more lines based on each method.